### PR TITLE
Bump github.com/golangci/golangci-lint from 1.40.1 to 1.41.1

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: extension/bearertokenauthextension/bearertokenauth_test.go [hunk 1]
```diff
@@ -54,9 +53,6 @@ func TestBearerAuthenticator(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, credential)
 
-	_, ok := credential.(credentials.PerRPCCredentials)
-	assert.True(t, ok)
```
Comment (tigrannajaryan) [inline extension/bearertokenauthextension/bearertokenauth_test.go:58]: Why is this deleted?

Comment (bogdandrutu) [reply extension/bearertokenauthextension/bearertokenauth_test.go:58]: because the return `credential` is of a type `credentials.PerRPCCredentials` so this check is useless, always will be that, we nolonger return an interface so no check is needed it is ensured by the compiler.

**Comment by bogdandrutu** [comment]
@tigrannajaryan PTAL

**Review by tigrannajaryan — APPROVED** [review]
